addons/binding/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/SensorThingHandler.java

```diff
@@ -188,10 +182,43 @@ public void bridgeStatusChanged(@NonNull ThingStatusInfo bridgeStatusInfo) {
                     }
 
                     // Some sensors support optional channels
+                    // (see https://github.com/dresden-elektronik/deconz-rest-plugin/wiki/Supported-Devices#sensors)
+                    // any battery-powered sensor
+                    Integer batteryLevel = newState.config.battery;
+                    if (batteryLevel != null) {
+                        ignoreConfigurationUpdate = true;
+                        createAndUpdateChannelIfExists(CHANNEL_BATTERY_LEVEL,
+                                new DecimalType(batteryLevel.longValue()));
+                        createAndUpdateChannelIfExists(CHANNEL_BATTERY_LOW,
+                                batteryLevel <= 10 ? OnOffType.ON : OnOffType.OFF);
+                        ignoreConfigurationUpdate = false;
+                    }
+
+                    // some Xiaomi sensors
+                    Float temperature = newState.config.temperature;
+                    if (temperature instanceof Float) {
+                        createAndUpdateChannelIfExists(CHANNEL_TEMPERATURE,
+                                new QuantityType<Temperature>(temperature / 100, SIUnits.CELSIUS));
+                    }
+
+                    // ZHAPresence - e.g. IKEA TRÅDFRI motion sensor
                     if (newState.state.dark instanceof Boolean) {
                         createChannel(CHANNEL_DARK);
                     }
 
+                    // ZHAConsumption - e.g Bitron 902010/25 or Heiman SmartPlug
+                    if (newState.state.power instanceof Float) {
```

Comment:
Why is that `instanceof` and not `!= null`, am I missing something?

Reply:
Nothing specific. I first thought this is the better condition, but I does not matter. I can change it to `!= null` if you prefer.

Reply:
In early java editions `instanceof` had a considerable runtime cost. I guess that's not the case anymore. But why making the runtime more effort if not necessary. And consistency is an argument, as `!= null` is used in other cases.

Reply:
Agreed. Done.